pkg/provision/providers/vm/disk.go

```diff
@@ -49,15 +50,20 @@ func (p *Provisioner) CreateDisks(state *State, nodeReq provision.NodeRequest) (
 
 		defer diskF.Close() //nolint:errcheck
 
-		err = diskF.Truncate(int64(disk.Size))
+		if err = diskF.Truncate(int64(disk.Size)); err != nil {
+			return nil, err
+		}
+
+		if err = syscall.Fallocate(int(diskF.Fd()), 0, 0, int64(disk.Size)); err != nil {
+			fmt.Fprintf(os.Stderr, "WARNING: failed to allocate disk space for %q (size %d): %s", diskPath, disk.Size, err)
```

Comment:
Maybe `preallocate` will sound better here, since allocate means we can't get required storage at all, and thats not the case I think.

Reply:
good point, fixed